src/Cake/Commands/BootstrapCommand.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cake.Core;
+using Cake.Core.Configuration;
+using Cake.Core.IO;
+using Cake.Core.Scripting;
+using Cake.Core.Scripting.Analysis;
+
+namespace Cake.Commands
+{
+    internal sealed class BootstrapCommand : ICommand
+    {
+        private readonly IScriptAnalyzer _analyzer;
+        private readonly ICakeConfiguration _configuration;
+        private readonly IScriptProcessor _processor;
+        private readonly ICakeEnvironment _environment;
+
+        // Delegate factory used by Autofac.
+        public delegate BootstrapCommand Factory();
+
+        public BootstrapCommand(
+            IScriptAnalyzer analyzer,
+            ICakeConfiguration configuration,
+            IScriptProcessor processor,
+            ICakeEnvironment environment)
+        {
+            _analyzer = analyzer;
+            _configuration = configuration;
+            _processor = processor;
+            _environment = environment;
+        }
+
+        public bool Execute(CakeOptions options)
+        {
+            // Get the script path and the root.
+            var path = options.Script.MakeAbsolute(_environment);
+            var root = path.GetDirectory();
+
+            // Analyze the script.
+            var result = _analyzer.Analyze(path);
```

Comment:
@patriksvensson the result variable is not being used.  Should it be?

Reply:
@gep13 it is used. See next call, `result.Modules`

Reply:
It's used but it should also check `result.Succeeded` if it's false it shouldn't continue to install modules. But it should abort, error log `result.Errors` and exit with a non zero code.

Something like:
```csharp
if (!result.Succeeded)
{
    throw new AggregateException(
        $"Bootstrap command failed to analyze script: {path}",
        result.Errors.Select(
            error => new CakeException($"File: {error.File}, Line: {error.Line}, Message: {error.Message}")));
}
```

Currently it just silently fails and reports success, which Isn't what users would expect.